### DISCUSSION TITLE
ssh tunnel

### DISCUSSION THREAD

**Discussion by themayna** [deployphp/deployer#3690] [Help needed]
i love this script and have use it many times before
but now, i'm in a situation where i have a jump server -> stage server that i just can't configure
please let me know if there is any way i can "hack" the ssh command to work

**Comment by acharseth** [comment]
I guess you could set up SSH tunneling, if you are allowed to, making this work but a more secure way to do this is to split the deploy task into two part. One for building a tar or .phar-file, upload it to a shared sftp server and then run the second step deploy script on the server collection the tar/.phar file and deploying it.

**Comment by paolooo** [comment]
Have you tried using `ssh -A -J jump@server stage@server` which you can do an equivalent deployer ssh configure with `host(...)->set('ssh_argument', ...)`

See:
https://www.tecmint.com/access-linux-server-using-a-jump-host/
https://deployer.org/docs/7.x/hosts
